usr/share/rear/skel/default/etc/scripts/run-sshd

```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
 # check inittab for "ssh:23:respawn:/bin/sshd -D"
-if $(grep -q ^ssh: /etc/inittab) ; then
-    if ! ls /etc/ssh/*key* &>/dev/null ; then
-        # generate at least one SSH host key if there is none because
-        # a running sshd without any SSH host key is not accessible from remote
-        # (on the remote host one gets "Connection to recovery.system.IP.address closed.")
+if grep -q '^ssh:' /etc/inittab ; then
+    if ! test -s /etc/ssh/ssh_host_rsa_key ; then
+        # generate at least an rsa SSH host key if there is none to be fail-safe against
+        # running sshd possibly without any SSH host key which is not accessible from remote
+        # (on the remote host one would get "Connection to recovery.system.IP.address closed.")
         # cf. build/default/500_ssh_setup.sh
         ssh-keygen -q -t rsa -N '' -f /etc/ssh/ssh_host_rsa_key
```

Comment:
The host key just generated could be displayed on the system console by adding this line directly below `ssh-keygen -q -t rsa ...`:
```
        echo -e "\nSSH fingerprint: $(ssh-keygen -l -f /etc/ssh/ssh_host_rsa_key.pub)\n" >> /etc/issue
```
In this case, a user logging in via SSH would have a chance to verify that he is connecting to the right host, assuming the user has console access (e.g. via a KVM connection).

Reply:
And I think the '-q' is also not needed here for ssh-keygen.